internal/resource/config/passwordcredentialvalidator/password_credential_validator_resource.go

```diff
@@ -122,6 +122,119 @@ func (r *passwordCredentialValidatorResource) Schema(ctx context.Context, req re
+		if !hasClientId || clientIdCount == 0 {
+			resp.Diagnostics.AddError("The \"Client Id\" field is required for the PingOne for Enterprise Directory Password Credential Validator", "")
+		}
+		hasClientSecret, clientSecretCount := internaltypes.ValidateStringCountInSlice(fieldNameValues, "Client Secret", 1)
+		if !hasClientSecret || clientSecretCount == 0 {
+			resp.Diagnostics.AddError("The \"Client Secret\" field is required for the PingOne for Enterprise Directory Password Credential Validator", "")
+		}
+
+	case "com.pingidentity.plugins.pcvs.p14c.PingOneForCustomersPCV":
+		hasPingOneForCustomersDs, pingOneForCustomersDsCount := internaltypes.ValidateStringCountInSlice(fieldNameValues, "PingOne for Customers Datastore", 1)
+		if !hasPingOneForCustomersDs || pingOneForCustomersDsCount == 0 {
+			resp.Diagnostics.AddError("The \"PingOne for Customers Datastore\" field is required for the PingOne Password Credential Validator", "")
+		}
+
+	case "com.pingidentity.plugins.pcvs.pingid.PingIdPCV":
+		hasAuthenticationDuringErrors, authenticationDuringErrorsCount := internaltypes.ValidateStringCountInSlice(fieldNameValues, "Authentication During Errors", 1)
+		if !hasAuthenticationDuringErrors || authenticationDuringErrorsCount == 0 {
+			resp.Diagnostics.AddError("The \"Authentication During Errors\" field is required for the PingID Password Credential Validator", "")
+		}
+
+	case "org.sourceid.saml20.domain.LDAPUsernamePasswordCredentialValidator":
+		hasLdapDs, ldapDsCount := internaltypes.ValidateStringCountInSlice(fieldNameValues, "LDAP Datastore", 1)
+		if !hasLdapDs || ldapDsCount == 0 {
+			resp.Diagnostics.AddError("The \"LDAP Datastore\" field is required for the LDAP Username Password Credential Validator", "")
+		}
+		hasSearchBase, searchBaseCount := internaltypes.ValidateStringCountInSlice(fieldNameValues, "Search Base", 1)
+		if !hasSearchBase || searchBaseCount == 0 {
+			resp.Diagnostics.AddError("The \"Search Base\" field is required for the LDAP Username Password Credential Validator", "")
+		}
+		hasSearchFilter, searchFilterCount := internaltypes.ValidateStringCountInSlice(fieldNameValues, "Search Filter", 1)
+		if !hasSearchFilter || searchFilterCount == 0 {
+			resp.Diagnostics.AddError("The \"Search Filter\" field is required for the LDAP Username Password Credential Validator", "")
+		}
+
+	case "org.sourceid.saml20.domain.RadiusUsernamePasswordCredentialValidator":
+		hasRadiusDs, radiusDsCount := internaltypes.ValidateStringCountInSlice(tableNames, "RADIUS Servers", 1)
+		if !hasRadiusDs || radiusDsCount == 0 {
+			resp.Diagnostics.AddError("At least one \"RADIUS Servers\" table is required for the RADIUS Username Password Credential Validator", "")
+		}
+
+	case "org.sourceid.saml20.domain.SimpleUsernamePasswordCredentialValidator":
+		hasCorrectUsernameCount, usernameCount := internaltypes.ValidateStringCountInSlice(nestedTableFieldNameValues, "Username", rowCount)
```

Comment:
For these ones that are expected certain fields in every single row, I might just move that check up to the part where you are looping through them.